tests/testsuite/build.rs

```diff
@@ -1676,6 +1676,154 @@ Caused by:
+        .file("src/lib.rs", "")
+        .file(
+            "a/Cargo.toml",
+            r#"
+        [package]
+        name = "a"
+        version = "0.0.1"
+        "#,
+        )
+        .file("a/src/lib.rs", "")
+        .build();
+    p.cargo("build")
+    .with_stderr_contains(
+        "[WARNING] found both `dev-dependencies` and `dev_dependencies` are set in the `foo` package",
+    )
+        .run();
+}
+
+#[cargo_test]
+fn build_dependencies_conflicting_warning() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+        [package]
+        name = "foo"
+        version = "0.1.0"
+        edition = "2018"
+
+        [build-dependencies]
+        a = {path = "a"}
+        [build_dependencies]
+        a = {path = "a"}
+        "#,
+        )
+        .file("src/lib.rs", "")
+        .file(
+            "a/Cargo.toml",
+            r#"
+        [package]
+        name = "a"
+        version = "0.0.1"
+        "#,
+        )
+        .file("a/src/lib.rs", "")
+        .build();
+    p.cargo("build")
+    .with_stderr_contains(
+        "[WARNING] found both `build-dependencies` and `build_dependencies` are set in the `foo` package",
+    )
+        .run();
+}
+
+#[cargo_test]
+fn lib_crate_types_conflicting_warning() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [project]
+                name = "foo"
+                version = "0.5.0"
+                authors = ["wycats@example.com"]
+
+                [lib]
+                name = "foo"
+                crate-type = ["rlib", "dylib"]
+                crate_type = ["staticlib", "dylib"]
+            "#,
+        )
+        .file("src/lib.rs", "pub fn foo() {}")
+        .build();
+    p.cargo("build")
+    .with_stderr_contains(
+        "[WARNING] found both `crate-type` and `crate_type` are set in the `foo` library target",
+    )
+        .run();
+}
+
+#[cargo_test]
+fn examples_crate_types_conflicting_warning() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [project]
+                name = "foo"
+                version = "0.5.0"
+                authors = ["wycats@example.com"]
+
+                [[example]]
+                name = "ex"
+                path = "examples/ex.rs"
+                crate-type = ["rlib", "dylib"]
+                crate_type = ["proc_macro"]
+                [[example]]
+                name = "goodbye"
+                path = "examples/ex-goodbye.rs"
+                crate-type = ["rlib", "dylib"]
+                crate_type = ["rlib", "staticlib"]
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .file(
+            "examples/ex.rs",
+            r#"
+            extern crate proc_macro;
+            use proc_macro::TokenStream;
+
+            #[proc_macro]
+            pub fn eat(_item: TokenStream) -> TokenStream {
+                "".parse().unwrap()
+            }
+            "#,
```

Comment:
I will simplify it, but we should probably keep the file content, otherwise it will look a bit  strange. Most of the other tests have content as well.